app/services/hubee_api_client.rb

```diff
@@ -0,0 +1,89 @@
+class HubEEAPIClient
+  class AlreadyExistsError < StandardError; end
+
+  def get_organization(siret, code_commune)
+    faraday_connection.get { |req|
+      req.url "#{api_host}/referential/v1/organizations/SI-#{siret}-#{code_commune}"
+      req.headers['Authorization'] = "Bearer #{access_token}"
+      req.headers['Content-Type'] = 'application/x-www-form-urlencoded'
+    }.body.with_indifferent_access
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  def create_organization(organization_payload)
+    faraday_connection.post { |req|
+      req.url "#{api_host}/referential/v1/organizations"
+      req.headers['Authorization'] = "Bearer #{access_token}"
+      req.headers['Content-Type'] = 'application/json'
+      req.headers['tag'] = 'Portail HubEE'
+      req.body = organization_payload.to_json
+    }.body.with_indifferent_access
+  rescue Faraday::BadRequestError => e
+    raise AlreadyExistsError if e.response[:body]['errors'][0]['message'].include?('already exists')
+
+    raise
+  end
+
+  # rubocop:enable Metrics/AbcSize
+
+  # rubocop:disable Metrics/AbcSize
+  def create_subscription(subscription_body)
+    faraday_connection.post { |req|
+      req.headers['Authorization'] = "Bearer #{access_token}"
+      req.headers['Content-Type'] = 'application/json'
+      req.url "#{api_host}/referential/v1/subscriptions"
+      req.body = subscription_body.to_json
+    }.body.with_indifferent_access
+  rescue Faraday::BadRequestError => e
+    raise AlreadyExistsError if e.response[:body]['errors'][0]['message'].include?('already exists')
+
+    raise
+  end
+
+  # rubocop:enable Metrics/AbcSize
+
+  private
+
+  def access_token
+    response = faraday_connection.post do |req|
+      req.url hubee_auth_url
+      req.headers['Authorization'] = "Basic #{encoded_client_id_and_secret}"
+      req.headers['Content-Type'] = 'application/x-www-form-urlencoded'
+      req.headers['tag'] = 'Portail HubEE'
+      req.body = URI.encode_www_form({ grant_type: 'client_credentials', scope: 'ADMIN' })
+    end
+
+    response.body['access_token']
```

Comment:
```suggestion
    return @access_token if @access_token.present?
    
    response = faraday_connection.post do |req|
      req.url hubee_auth_url
      req.headers['Authorization'] = "Basic #{encoded_client_id_and_secret}"
      req.headers['Content-Type'] = 'application/x-www-form-urlencoded'
      req.headers['tag'] = 'Portail HubEE'
      req.body = URI.encode_www_form({ grant_type: 'client_credentials', scope: 'ADMIN' })
    end

    @access_token = response.body['access_token']
```

Ça évite de doubler les appels et de récupérer uniquement un jeton.
L'alternative est de faire comme pour le client INSEE.